src/main/java/org/dataone/hashstore/filehashstore/FileHashStore.java

```diff
@@ -540,266 +508,66 @@ private ObjectMetadata syncPutObject(
+            unTagObject(pid, cid);
+            throw e;
 
         } finally {
             // Release lock
-            synchronized (referenceLockedCids) {
-                logFileHashStore.debug(
-                    "FileHashStore.tagObject - Releasing referenceLockedCids for cid: " + cid
-                );
-                referenceLockedCids.remove(cid);
-                referenceLockedCids.notifyAll();
-            }
-        }
-    }
-
-    @Override
-    public String findObject(String pid) throws NoSuchAlgorithmException, IOException {
-        logFileHashStore.debug("FileHashStore.findObject - Called to find object for pid: " + pid);
-        FileHashStoreUtility.ensureNotNull(pid, "pid", "findObject");
-        FileHashStoreUtility.checkForEmptyString(pid, "pid", "findObject");
-
-        // Get path of the pid references file
-        Path absPidRefsPath = getRealPath(pid, "refs", "pid");
-
-        if (Files.exists(absPidRefsPath)) {
-            String cid = new String(Files.readAllBytes(absPidRefsPath));
-            Path absCidRefsPath = getRealPath(cid, "refs", "cid");
-
-            // Throw exception if the cid refs file doesn't exist
-            if (!Files.exists(absCidRefsPath)) {
-                String errMsg =
-                    "FileHashStore.deleteObject - Cid refs file does not exist for cid: " + cid
-                        + " with address: " + absCidRefsPath + ", but pid refs file exists.";
-                logFileHashStore.error(errMsg);
-                throw new OrphanPidRefsFileException(errMsg);
-            }
-            // If the pid is found in the expected cid refs file, return it
-            if (isPidInCidRefsFile(pid, absCidRefsPath)) {
-                logFileHashStore.info(
-                    "FileHashStore.findObject - Cid (" + cid + ") found for pid:" + pid
-                );
-                return cid;
-
-            } else {
-                String errMsg = "FileHashStore.deleteObject - Pid refs file exists, but pid (" + pid
-                    + ") not found in cid refs file for cid: " + cid + " with address: "
-                    + absCidRefsPath;
-                logFileHashStore.error(errMsg);
-                throw new PidNotFoundInCidRefsFileException(errMsg);
-            }
-
-        } else {
-            String errMsg = "FileHashStore.findObject - Unable to find cid for pid: " + pid
-                + ". Pid refs file does not exist at: " + absPidRefsPath;
-            logFileHashStore.error(errMsg);
-            // Create custom exception class
-            throw new FileNotFoundException(errMsg);
+            releaseReferencedLockedCids(cid);
         }
     }
 
     @Override
     public String storeMetadata(InputStream metadata, String pid, String formatId)
-        throws IOException, FileNotFoundException, IllegalArgumentException, InterruptedException,
+        throws IOException, IllegalArgumentException, FileNotFoundException, InterruptedException,
```

Comment:
Close input stream

Reply:
Thank you! I have added the try-finally blocks to do so.